### PR TITLE
fix: elaborate on "non-flux state management"

### LINKED ISSUE #1224: Style Guide: Elaborate on “Non-Flux State Management”
I think it's better to elaborate a little bit on this because “Flux” appears in the section title and is not discussed later in the content.

### PR DESCRIPTION
**Pull request by sarahdayan**
This PR clarifies what Flux is and what its relationship with Vuex is, adds a link to the relevant part of the docs on that subject, and briefly explain its benefits.

fix #1224

### CODE REVIEW

File: src/v2/style-guide/index.md [hunk 1]
```diff
@@ -1941,7 +1941,9 @@ Vue.component('TodoItem', {
 
 **[Vuex](https://github.com/vuejs/vuex) should be preferred for global state management, instead of `this.$root` or a global event bus.**
 
-Managing state on `this.$root` and/or using a [global event bus](https://vuejs.org/v2/guide/migration.html#dispatch-and-broadcast-replaced) can be convenient for very simple cases, but are not appropriate for most applications. Vuex offers not only a central place to manage state, but also tools for organizing, tracking, and debugging state changes.
+Managing state on `this.$root` and/or using a [global event bus](https://vuejs.org/v2/guide/migration.html#dispatch-and-broadcast-replaced) can be convenient for very simple cases, but are not appropriate for most applications.
```
Comment (bencodezen) [inline src/v2/style-guide/index.md]: I know this isn't your original sentence, but could you update this to match the subject plurality

```suggestion
Managing state on `this.$root` and/or using a [global event bus](https://vuejs.org/v2/guide/migration.html#dispatch-and-broadcast-replaced) can be convenient for very simple cases, but it is not appropriate for most applications.
```

**Comment by bencodezen** [comment]
Thanks for the quick change @sarahdayan! 

@NataliaTepluhina going to wait for your review just to make sure I didn't miss anything, but I think we're good to go!

File: src/v2/style-guide/index.md [hunk 2]
```diff
@@ -1941,7 +1941,9 @@ Vue.component('TodoItem', {
 
 **[Vuex](https://github.com/vuejs/vuex) should be preferred for global state management, instead of `this.$root` or a global event bus.**
 
-Managing state on `this.$root` and/or using a [global event bus](https://vuejs.org/v2/guide/migration.html#dispatch-and-broadcast-replaced) can be convenient for very simple cases, but are not appropriate for most applications. Vuex offers not only a central place to manage state, but also tools for organizing, tracking, and debugging state changes.
+Managing state on `this.$root` and/or using a [global event bus](https://vuejs.org/v2/guide/migration.html#dispatch-and-broadcast-replaced) can be convenient for very simple cases, but it is not appropriate for most applications.
+
+Vuex is the [official flux-like implementation](https://vuejs.org/v2/guide/state-management.html#Official-Flux-Like-Implementation) for Vue, and offers not only a central place to manage state, but also tools for organizing, tracking, and debugging state changes. It simplifies debugging, and integrates well in the Vue ecosystem (including full [Vue DevTools](https://vuejs.org/v2/guide/installation.html#Vue-Devtools) support).
```
Comment (NataliaTepluhina) [inline src/v2/style-guide/index.md]: @sarahdayan I think we are repeating `debugging` twice here; maybe let's remove the second mention?
```suggestion
Vuex is the [official flux-like implementation](https://vuejs.org/v2/guide/state-management.html#Official-Flux-Like-Implementation) for Vue, and offers not only a central place to manage state, but also tools for organizing, tracking, and debugging state changes. It integrates well in the Vue ecosystem (including full [Vue DevTools](https://vuejs.org/v2/guide/installation.html#Vue-Devtools) support).
```

Comment (sarahdayan) [reply src/v2/style-guide/index.md]: Great point!

**Pull request merged**